addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rotelra1x.handler;
+
+import static org.openhab.binding.rotelra1x.RotelRa1xBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.rotelra1x.internal.ConfigurationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.PortInUseException;
+import gnu.io.RXTXPort;
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link RotelRa1xHandler} handles commands and state changes, and communicates
+ * with the amplifier over the serial port interface..
+ *
+ * @author Marius Bjørnstad - Initial contribution
+ */
+public class RotelRa1xHandler extends BaseThingHandler {
+
+    private static final int BAUD = 115200;
+    private static final long ERROR_RETRY_DELAY_MS = 60000;
+    private int maximumVolume;
+    private RXTXPort serialPort;
+
+    private boolean exit;
+    private volatile boolean power;
+
+    private final Logger logger = LoggerFactory.getLogger(RotelRa1xHandler.class);
+
+    public RotelRa1xHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        maximumVolume = ((BigDecimal) getThing().getConfiguration().get("maximum-volume")).intValue();
+        exit = false;
+        try {
+            connect();
+            Executors.newSingleThreadScheduledExecutor().schedule(() -> {
```

Comment:
I saw the comment from @kaikreuzer  to use `Executors.newSingleThreadScheduledExecutor()` but given the code style recommendations to not create own threads, but use scheduler or the thread pool I'm somewhat confused.

Reply:
You might want to use `Executors.newSingleThreadExecutor().submit()` instead, but then it is imho ok to create and block a custom thread this way.

Reply:
I happened to have to make some changes here. This is not addressed

Reply:
Good that this is OK, I will leave it. It would be possible to construct a non-blocking solution by first scheduling the 4 second delay in the shared thread pool, then having the callback start a custom thread. This would however be a bit complex.

The larger issue is perhaps the reason for this arbitrary delay, which looks like a race condition in disguise. If the input loop is started immediately, then the updateState() calls do not work. Is it ok to leave it, seeing as it works well in practice?